server/controllers/notificationController.ts

```diff
@@ -0,0 +1,155 @@
+        errors,
+        data,
+        csrfToken,
+        moment,
+        employeeName,
+        authUrl,
+      })
+    } catch (error) {
+      return next(error)
+    }
+  }
+
+  async resume(req: Request, res: Response, next: NextFunction) {
+    try {
+      const {
+        user: { token },
+        app,
+      } = req
+      const {
+        notificationService: { resumeNotifications },
+      } = app.get('DataServices')
+      await resumeNotifications(token)
+      res.redirect('back')
+    } catch (error) {
+      serviceUnavailable(req, res)
+    }
+  }
+
+  async getManage(req: Request, res: Response, next: NextFunction) {
+    try {
+      const {
+        user: { employeeName, token },
+        app,
+        authUrl,
+      } = req
+
+      const {
+        locals: { csrfToken, errors = null },
+      } = res
+      const { notificationService } = app.get('DataServices')
+      const { snoozeUntil: rawSnoozeUntil, preference = NONE } = await notificationService.getPreferences(token)
+      const notificationsEnabled = preference !== NONE
+
+      return res.render('pages/manage-your-notifications', {
+        errors,
+        csrfToken,
+        notificationsEnabled,
+        snoozeUntil: notificationsEnabled ? getSnoozeUntil(rawSnoozeUntil) : '',
+        employeeName,
+        authUrl,
+      })
+    } catch (error) {
+      return next(error)
+    }
+  }
+
+  async getPause(req: Request, res: Response, next: NextFunction) {
+    try {
+      const {
+        user: { employeeName, token },
+        query: { pauseUnit, pauseValue },
+        app,
+        authUrl,
+      } = req
+
+      const {
+        locals: { csrfToken, errors },
+      } = res
+      const { notificationService } = app.get('DataServices')
+      const { snoozeUntil: rawSnoozeUntil, preference = NONE } = await notificationService.getPreferences(token)
+      const notificationsEnabled = preference !== NONE
+
+      return res.render('pages/pause-notifications', {
+        errors,
+        csrfToken,
+        notificationsEnabled,
+        snoozeUntil: notificationsEnabled ? getSnoozeUntil(rawSnoozeUntil) : '',
+        employeeName,
+        authUrl,
+        pauseUnit,
+        pauseValue,
+      })
+    } catch (error) {
+      return next(error)
+    }
+  }
+
+  async setPause(req: Request, res: Response, next: NextFunction) {
+    try {
+      const errors = validationResult(req)
+      res.locals.errors = errors
+      const {
+        user: { token },
+        app,
+        body: { pauseUnit, pauseValue },
+      } = req
+
+      if (errors.isEmpty()) {
+        const {
+          notificationService: { updateSnooze },
+        } = app.get('DataServices')
+        await updateSnooze(token, getFormattedFutureDate(pauseUnit, pauseValue))
+        return res.redirect('/notifications/manage')
+      }
+
+      req.query = { pauseUnit, pauseValue }
+      return await new NotificationController().getPause(req, res, next)
```

Comment:
this is a bit naff.  For some reason `this` comes out as `undefined` so had to create the new controller to then chain through to the `getPause` method.  The original chained through express which was quite confusing and not clear what was going on